zerver/actions/message_edit.py

```diff
@@ -430,6 +430,111 @@ def get_visibility_policy_after_merge(
     return UserTopic.VisibilityPolicy.INHERIT
 
 
+def update_message_content(
+    user_profile: UserProfile,
+    target_message: Message,
+    content: str,
+    rendering_result: MessageRenderingResult,
+    prior_mention_user_ids: set[int],
+    mention_data: MentionData,
+    event: dict[str, Any],
+    edit_history_event: EditHistoryEvent,
+    stream_topic: StreamTopicTarget | None,
+) -> None:
+    """
+    The main function for message editing.  A message edit event can
+    modify:
+    * the message's content (in which case the caller will have
+      set both content and rendered_content),
+    * the topic, in which case the caller will have set topic_name
+    * or both message's content and the topic
+    * or stream and/or topic, in which case the caller will have set
+        new_stream and/or topic_name.
+
+    With topic edits, propagate_mode determines whether other message
+    also have their topics edited.
+    """
```

Comment:
This docstring seems misplaced, if this the content code path?

Reply:
Fixed this.